### PR TITLE
Implement a Mhelp ticket system. Also fixes a bug in the old one

### PR DESCRIPTION
**Pull request by farie82**
**What does this PR do:**
Adds a mhelp ticket system. Works like the ahelp ticket system.
Take in the message now actually takes the question using the ticket system.
A follow up question now gets assigned to the last M/A-help. Was a bug in the code
Mentors can now also close mhelps and it'll give a more fitting message.
The My Tickets verb is now public to all users. I don't know why it was locked behind an admin right check. But hey now players can see their open tickets.

fixes: #10976

How it looks: 
image

**Changelog:**
:cl:
add: Adds a Mhelp ticket system
tweak: Take in the m/a-helps now actually takes the question
fix: a/m-help followup responses using a-mhelp verb now get added to the last ticket like it should
fix: Normal players can now see their open tickets using the My Admin/Mentor tickets verb
/:cl:

### CODE REVIEW

File: code/__DEFINES/preferences.dm [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #define CHAT_NO_TICKETLOGS 65536
 #define UI_DARKMODE 131072
 #define DISABLE_KARMA 262144
+#define CHAT_NO_MENTORTICKETLOGS 524288
```
Comment (Fox-McCloud) [inline code/__DEFINES/preferences.dm]: This is ***disgusting***

At this point, we're abusing flags.

Comment (farie82) [reply code/__DEFINES/preferences.dm]: Aye it's gross. Can do it in another way but hey it works

Comment (AffectedArc07) [reply code/__DEFINES/preferences.dm]: image

**Comment by AffectedArc07** [comment]
https://github.com/ParadiseSS13/Paradise/blob/e57f0b9306060be7ecd6177db7e5a674adbc96ce/code/modules/client/preference/preferences_mysql.dm#L58 You need to increase this
Replace `524287` with `1048575`

**Comment by AffectedArc07** [comment]
__SQL Verification__

This PR has been SQL-tested. Toggle now works.

**Comment by farie82** [comment]
> **SQL Verification**
> 
> This PR has been SQL-tested. Toggle now works.

Luv u. <3
No but thanks for testing. Saves me a bunch of trouble

File: code/controllers/subsystem/tickets/tickets.dm [hunk 2]
```diff
@@ -326,6 +334,10 @@ UI STUFF
 		to_chat(target, text)
 	return TRUE
 
+//Sends a message to the designated staf
+/datum/controller/subsystem/tickets/proc/message_staf(var/msg)
```
Comment (Ty-Omaha) [inline code/controllers/subsystem/tickets/tickets.dm]: staff is spelt with 2 F's

Comment (farie82) [reply code/controllers/subsystem/tickets/tickets.dm]: one mistake to rule them all. Fixed now

File: code/__DEFINES/preferences.dm (hunk 1 above)
Comment (Citinited) [reply code/__DEFINES/preferences.dm]: If they didn't want us to abuse it, why would they let us abuse it? 

😄

**Comment by variableundefined** [comment]
This PR will take a bit to resolve as Fox and I need to talk about the usage of preferences flags, since they will eventually run out.The only alternative would be using list, but that would require redoing it on the database side.

**Comment by AffectedArc07** [comment]
DB Flag analysis. There are 4 ways of storing an integer in a MySQL database, 5 if you count text. The types are as follows:

BIGINT: 9,223,372,036,854,775,807 (Can store 63 flags)
INT: 2,147,483,647 (Can store 31 flags) | This is what we currently use
SMALLINT: 32,767 (Can store 15 flags)
TINYINT: 255 (Can store 7 flags)

In our code we currently use 18 flags, with 13 to spare

I am considering refactoring flags to use lists or JSON instead, but dont know if there is demand. Maints can I get a yay or nay

**Comment by variableundefined** [comment]
@farie82 Conflicts

**Pull request merged**